lib/core/buckets/usecase.ts

```diff
@@ -213,6 +244,157 @@ export class BucketsUsecase {
     return (usage && usage.total) || 0;
   } 
 
+  async startUpload(
+    userId: string,
+    bucketId: string,
+    cluster: string[],
+    uploads: { index: number; size: number }[],
+    auth: { username: string; password: string },
+    multiparts: number = 1
+  ) {
+    const [bucket, user] = await Promise.all([
+      this.bucketsRepository.findOne({ id: bucketId }),
+      this.usersRepository.findById(userId),
+    ]);
+
+    if (!user) {
+      throw new UserNotFoundError();
+    }
+
+    if (!bucket) {
+      throw new BucketNotFoundError();
+    }
+
+    if (bucket.user !== userId) {
+      throw new BucketForbiddenError();
+    }
+
+    if (!uploads) {
+      throw new MissingUploadsError();
+    }
+
+    if (!Array.isArray(uploads)) {
+      throw new InvalidUploadsError();
+    }
+
+    const uploadIndexesWithoutDuplicates = new Set(
+      uploads.map((upload) => upload.index)
+    );
+
+    if (uploadIndexesWithoutDuplicates.size < uploads.length) {
```

Comment:
Same here

Reply:
fixed

Reply:
In this one i'm not entirely sure we want to move it to the controller since statically we cannot validate this